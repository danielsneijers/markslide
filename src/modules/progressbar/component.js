// @flow
import React from 'react'
import CSS from './style.css'
import type { ProgressBarProps } from './types'

const ProgressBar = ({ offset }: ProgressBarProps): React$Element<any> => {
  const style = { transform: `translateX(-${offset}%)` }

  return <span className={CSS.progressBar} style={style} />
}

export default ProgressBar
