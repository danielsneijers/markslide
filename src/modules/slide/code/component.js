// @flow
import React from 'react'
import { compose } from 'ramda'
import classNames from 'classnames'
import { fullScreenOption, autoScaleContent } from 'higher-order-components'
import CSS from './style.css'

type Props = {
  index: number,
  content: string,
  meta: {
    class?: string // eslint-disable-line react/no-unused-prop-types
  },
  nextSlide: Function,
  style: {}
};

const CodeSlide = ({ content, index, meta, style, nextSlide }: Props) => {
  const slideClasses = classNames(
    CSS.slide,
    meta.class,
    'code-slide',
    `slide-${index}`
  )

  return (
    <div className={slideClasses} key={`slide-${index}`}>
      <div
        id={`slide-${index}`}
        key={Math.random()}
        className={CSS.code}
        style={{ opacity: 0, ...style }}
        onClick={nextSlide}
        dangerouslySetInnerHTML={{ __html: content }}
      />
    </div>
  )
}

const enhance = compose(fullScreenOption, autoScaleContent)

export default enhance(CodeSlide)
