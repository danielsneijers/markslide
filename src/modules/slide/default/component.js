// @flow
import React from 'react'
import { compose } from 'ramda'
import classNames from 'classnames'
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup'
import { fullScreenOption, autoScaleContent } from 'higher-order-components'
import CSS from './style.css'

type Props = {
  index: number,
  content: string,
  meta: {
    class?: string // eslint-disable-line react/no-unused-prop-types
  },
  nextSlide: Function,
  style: {}
};

const Slide = ({ content, index, nextSlide, meta, style }: Props) => {
  const slideClasses = classNames(
    CSS.slide,
    meta.class,
    'regular-slide',
    `slide-${index}`
  )

  return (
    <CSSTransitionGroup
      transitionName='slide'
      transitionAppearTimeout={500}
      transitionEnterTimeout={500}
      transitionLeaveTimeout={300}
      transitionAppear
      component='div'
      className={slideClasses}
      key={`slide-${index}`}
      onClick={nextSlide}
    >
      <div
        id={`slide-${index}`}
        style={{ opacity: 0, ...style }}
        dangerouslySetInnerHTML={{ __html: content }}
      />
    </CSSTransitionGroup>
  )
}

const enhance = compose(fullScreenOption, autoScaleContent)

export default enhance(Slide)
