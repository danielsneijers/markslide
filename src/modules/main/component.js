// @flow
import React, { PureComponent } from 'react'
import { Switch, Route, Redirect } from 'react-router'
import { applyTheme } from 'config/themes'
import { keyboardNavigation } from 'higher-order-components'
import { ProgressBarContainer } from 'modules/progressbar'
import CSS from './style.css'

type Props = {
  renderRouteContainer: () => React$Element<any>
};

export class Main extends PureComponent {
  props: Props;

  componentWillMount () {
    applyTheme()
  }

  render (): React$Element<any> {
    const { renderRouteContainer } = this.props

    return (
      <div className={CSS.wrapper}>
        <ProgressBarContainer />
        <Switch>
          <Route path='/:slide' render={renderRouteContainer} />
          <Redirect exact from='/' to='/1' />
        </Switch>
      </div>
    )
  }
}

export default keyboardNavigation(Main)
