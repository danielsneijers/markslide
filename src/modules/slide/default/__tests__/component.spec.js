import React from 'react'
import { shallow } from 'enzyme'
import { component as DefaultSlide } from '../component'

let tree
const componentProps = {
  index: 2,
  content: `<pre><code class="lang-javascript"><span class='code-row'>console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">'foo'</span><span class="token punctuation">)</span></span></code></pre>`,
  meta: {},
  nextSlide: jest.fn(),
  style: {}
}

describe('slide/default/component', () => {
  afterAll(() => {
    jest.resetAllMocks()
  })

  beforeEach(() => {
    tree = shallow(<DefaultSlide {...componentProps} />)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot()
  })

  it('navigates to next slide onClick', () => {
    expect(componentProps.nextSlide).not.toHaveBeenCalled()

    tree.find('.regular-slide').simulate('click')

    expect(componentProps.nextSlide).toHaveBeenCalled()
  })
})
