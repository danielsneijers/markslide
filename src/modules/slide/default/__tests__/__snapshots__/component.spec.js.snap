// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slide/default/component renders correctly 1`] = `
<CSSTransitionGroup
  className="slide regular-slide slide-2"
  component="div"
  onClick={[Function]}
  transitionAppear={true}
  transitionAppearTimeout={500}
  transitionEnter={true}
  transitionEnterTimeout={500}
  transitionLeave={true}
  transitionLeaveTimeout={300}
  transitionName="slide"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<pre><code class=\\"lang-javascript\\"><span class='code-row'>console<span class=\\"token punctuation\\">.</span><span class=\\"token function\\">log</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">'foo'</span><span class=\\"token punctuation\\">)</span></span></code></pre>",
      }
    }
    id="slide-2"
    style={
      Object {
        "opacity": 0,
      }
    }
  />
</CSSTransitionGroup>
`;
